src/gluonts/model/seq2seq/_mq_dnn_estimator.py

```diff
@@ -14,112 +14,151 @@
-    Seq2SeqEncoder,
-)
+from gluonts.block.encoder import HierarchicalCausalConv1DEncoder, RNNEncoder
 from gluonts.block.quantile_output import QuantileOutput
 from gluonts.core.component import validated
 from gluonts.trainer import Trainer
-
-# Relative imports
-from ._forking_estimator import ForkingSeq2SeqEstimator
+from gluonts.model.seq2seq._forking_estimator import ForkingSeq2SeqEstimator
 
 
-class MQDNNEstimator(ForkingSeq2SeqEstimator):
+class MQCNNEstimator(ForkingSeq2SeqEstimator):
     """
-    Intermediate base class for a Multi-horizon Quantile Deep Neural Network
-    (MQ-DNN), [WTN+17]_. The class fixes the decoder is a multi-quantile MLP.
-    Subclasses fix the encoder to be either a Convolutional Neural Network
-    (MQ-CNN) or a Recurrent Neural Network (MQ-RNN).
+    An :class:`MQDNNEstimator` with a Convolutional Neural Network (CNN) as an
+    encoder and a multi-quantile MLP as a decoder. Implements the MQ-CNN Forecaster, proposed in [WTN+17]_.
     """
 
     @validated()
     def __init__(
         self,
-        encoder: Seq2SeqEncoder,
-        context_length: Optional[int],
-        prediction_length: int,
         freq: str,
-        # FIXME: why do we have two parameters here?
-        mlp_final_dim: int = 20,
-        mlp_hidden_dimension_seq: List[int] = list(),
-        quantiles: List[float] = list(),
+        prediction_length: int,
+        context_length: Optional[int] = None,
+        use_feat_dynamic_real: bool = False,
+        use_feat_static_cat: bool = False,
+        cardinality: List[int] = None,
+        embedding_dimension: List[int] = None,
+        add_time_feature: bool = False,
+        add_age_feature: bool = False,
+        seed: Optional[int] = None,
+        decoder_mlp_dim_seq: Optional[List[int]] = None,
+        channels_seq: Optional[List[int]] = None,
+        dilation_seq: Optional[List[int]] = None,
+        kernel_size_seq: Optional[List[int]] = None,
+        use_residual: bool = True,
+        quantiles: Optional[List[float]] = None,
         trainer: Trainer = Trainer(),
     ) -> None:
-        context_length = (
-            prediction_length if context_length is None else context_length
-        )
-        assert all(
-            [d > 0 for d in mlp_hidden_dimension_seq]
+
+        assert (
+            prediction_length > 0
+        ), f"Invalid prediction length: {prediction_length}."
+        assert decoder_mlp_dim_seq is None or all(
+            d > 0 for d in decoder_mlp_dim_seq
         ), "Elements of `mlp_hidden_dimension_seq` should be > 0"
+        assert channels_seq is None or all(
+            [d > 0 for d in channels_seq]
+        ), "Elements of `channels_seq` should be > 0"
+        assert dilation_seq is None or all(
+            [d > 0 for d in dilation_seq]
+        ), "Elements of `dilation_seq` should be > 0"
+        # TODO: add support for kernel size=1
+        assert kernel_size_seq is None or all(
+            [d > 1 for d in kernel_size_seq]
```

Comment:
Just an FYI, there is not need for `[ ]` here.

Reply:
thx!